parchmint/similaritymatcher.py

```diff
@@ -18,78 +18,95 @@ def __init__(
+        self._G2_param_diff_list = []
+        self._G1_layer_diff_list = []
+        self._G2_layer_diff_list = []
+        self._G1_port_diff_list = []
+        self._G2_port_diff_list = []
 
         super(SimilarityMatcher, self).__init__(G1.G, G2.G)
 
     def semantic_feasibility(self, G1_node, G2_node) -> bool:
-        """Returns True if adding (G1_node, G2_node) is symantically feasible.
-        The semantic feasibility function should return True if it is
-        acceptable to add the candidate pair (G1_node, G2_node) to the current
-        partial isomorphism mapping.   The logic should focus on semantic
-        information contained in the edge data or a formalized node class.
-        By acceptable, we mean that the subsequent mapping can still become a
-        complete isomorphism mapping.  Thus, if adding the candidate pair
-        definitely makes it so that the subsequent mapping cannot become a
-        complete isomorphism mapping, then this function must return False.
-        The default semantic feasibility function always returns True. The
-        effect is that semantics are not considered in the matching of G1
-        and G2.
-        The semantic checks might differ based on the what type of test is
-        being performed.  A keyword description of the test is stored in
-        self.test.  Here is a quick description of the currently implemented
-        tests::
-          test='graph'
-            Indicates that the graph matcher is looking for a graph-graph
-            isomorphism.
-          test='subgraph'
-            Indicates that the graph matcher is looking for a subgraph-graph
-            isomorphism such that a subgraph of G1 is isomorphic to G2.
-        Any subclass which redefines semantic_feasibility() must maintain
-        the above form to keep the match() method functional. Implementations
-        should consider multigraphs.
-        """
+      """Overriding semantic_feasibility to compare the layers, params, ports, and connections
+
+      Args:
+          G1_node (str): node 1
+          G2_node (str): node 2
+
+      Returns:
+          bool: if they are semantically feasible, return true. else return false. 
+      """
 
         # check each components. If not same, print out the difference and return false. 
-        feasible = True
+      feasible = True
 
 
-        G1_component = self.G1_device.get_component(G1_node)
-        G2_component = self.G2_device.get_component(G2_node)
+      G1_component = self._G1_device.get_component(G1_node)
+      G2_component = self._G2_device.get_component(G2_node)
 
+      # compare connectivities
+      for item in self._G1_in_edges:
+        conn_g1 = self._G1_in_edges[item]
+        conn_g2 = self._G2_in_edges[item]
 
-        if G1_component.layers != G2_component.layers:
-          print("layer wrong")
-          feasible = False
+        if conn_g1['source_port'].component != conn_g2['source_port'].component:
```

Comment:
when you `.component` you're only checking to see if its the right component. You need to check if `,port` is the same. It also doesn't account for the scenario when you have multiple connections between the same pair of components.